plugin/src/main/groovy/com/novoda/buildproperties/internal/LazyEntries.groovy

```diff
@@ -0,0 +1,34 @@
+package com.novoda.buildproperties.internal
+
+import com.novoda.buildproperties.Entries
+import com.novoda.buildproperties.Entry
+
+class LazyEntries extends Entries {
+
+    private final Closure<Entries> entriesClosure
+
+    LazyEntries(Closure<Entries> entriesClosure) {
+        this.entriesClosure = entriesClosure.memoize()
+    }
+
+    protected Entries getEntries() {
+        entriesClosure.call()
+    }
+
+    @Override
+    boolean contains(String key) {
+        entries.contains(key)
```

Comment:
is this some groovy magic where the getter gets called under the hood?

Reply:
yop. groovy supports implicit getters/setters for properties, so `getEntries()` is the same as reading the value of `entries`

Reply:
so confusing